instrumentation/spring/spring-webmvc/spring-webmvc-5.3/library/src/main/java/io/opentelemetry/instrumentation/spring/webmvc/v5_3/JavaxHttpServletRequestGetter.java

```diff
@@ -5,20 +5,39 @@
 
 package io.opentelemetry.instrumentation.spring.webmvc.v5_3;
 
-import io.opentelemetry.context.propagation.TextMapGetter;
-import java.util.Collections;
+import static java.util.Collections.emptyIterator;
+import static java.util.Collections.list;
+
+import io.opentelemetry.context.propagation.internal.ExtendedTextMapGetter;
+import java.util.Enumeration;
+import java.util.Iterator;
 import javax.servlet.http.HttpServletRequest;
 
-enum JavaxHttpServletRequestGetter implements TextMapGetter<HttpServletRequest> {
+enum JavaxHttpServletRequestGetter implements ExtendedTextMapGetter<HttpServletRequest> {
   INSTANCE;
 
   @Override
   public Iterable<String> keys(HttpServletRequest carrier) {
-    return Collections.list(carrier.getHeaderNames());
+    return list(carrier.getHeaderNames());
   }
 
   @Override
   public String get(HttpServletRequest carrier, String key) {
     return carrier.getHeader(key);
   }
+
+
+  @Override
+  public Iterator<String> getAll(HttpServletRequest carrier, String key) {
+    if (carrier == null) {
+      return emptyIterator();
+    }
```

Comment:
as far as I know non of the `TextMapGetter.get` implementations check for null carrier, I think we can skip it here too